egs/aishell2/ASR/prepare.sh

```diff
@@ -0,0 +1,141 @@
+#     - noise
+#     - speech
+
+dl_dir=$PWD/download
+
+. shared/parse_options.sh || exit 1
+
+# All files generated by this script are saved in "data".
+# You can safely remove "data" and rerun this script to regenerate it.
+mkdir -p data
+
+log() {
+  # This function is from espnet
+  local fname=${BASH_SOURCE[1]##*/}
+  echo -e "$(date '+%Y-%m-%d %H:%M:%S') (${fname}:${BASH_LINENO[0]}:${FUNCNAME[1]}) $*"
+}
+
+log "dl_dir: $dl_dir"
+
+if [ $stage -le 0 ] && [ $stop_stage -ge 0 ]; then
+  log "stage 0: Download data"
+
+  # If you have pre-downloaded it to /path/to/aishell2,
+  # you can create a symlink
+  #
+  #   ln -sfv /path/to/aishell2 $dl_dir/aishell2
+  #
+  # The directory structure is
+  # aishell2/
+  # |-- AISHELL-2
+  # |   |-- iOS
+  #         |-- data
+  #             |-- wav
+  #             |-- trans.txt
+  #         |-- dev
+  #             |-- wav
+  #             |-- trans.txt
+  #         |-- test
+  #             |-- wav
+  #             |-- trans.txt
+
+
+  # If you have pre-downloaded it to /path/to/musan,
+  # you can create a symlink
+  #
+  #   ln -sfv /path/to/musan $dl_dir/musan
+  #
+  if [ ! -d $dl_dir/musan ]; then
+    lhotse download musan $dl_dir
+  fi
+fi
+
+if [ $stage -le 1 ] && [ $stop_stage -ge 1 ]; then
+  log "Stage 1: Prepare aishell2 manifest"
+  # We assume that you have downloaded and unzip the aishell2 corpus
+  # to $dl_dir/aishell2
+  if [ ! -f data/manifests/.aishell_manifests.done ]; then
+    mkdir -p data/manifests
+    lhotse prepare aishell2 $dl_dir/aishell2 data/manifests -j $nj
+    touch data/manifests/.aishell2_manifests.done
+  fi
+fi
+
+if [ $stage -le 2 ] && [ $stop_stage -ge 2 ]; then
+  log "Stage 2: Prepare musan manifest"
+  # We assume that you have downloaded the musan corpus
+  # to data/musan
+  if [ ! -f data/manifests/.musan_manifests.done ]; then
+    log "It may take 6 minutes"
+    mkdir -p data/manifests
+    lhotse prepare musan $dl_dir/musan data/manifests
+    touch data/manifests/.musan_manifests.done
+  fi
+fi
+
+if [ $stage -le 3 ] && [ $stop_stage -ge 3 ]; then
+  log "Stage 3: Compute fbank for aishell"
+  if [ ! -f data/fbank/.aishell2.done ]; then
+    mkdir -p data/fbank
+    ./local/compute_fbank_aishell2.py
+    touch data/fbank/.aishell2.done
+  fi
+fi
+
+if [ $stage -le 4 ] && [ $stop_stage -ge 4 ]; then
+  log "Stage 4: Compute fbank for musan"
+  if [ ! -f data/fbank/.msuan.done ]; then
+    mkdir -p data/fbank
+    ./local/compute_fbank_musan.py
+    touch data/fbank/.msuan.done
+  fi
+fi
+
+if [ $stage -le 5 ] && [ $stop_stage -ge 5 ]; then
+  log "Stage 6: Prepare char based lang"
+  lang_char_dir=data/lang_char
+  mkdir -p $lang_char_dir
+
+  # Prepare text.
+  grep "\"text\":" data/manifests/aishell2_supervisions_train.jsonl.gz \
+    | sed -e 's/["text:\t ]*//g' | sed 's/,//g' \
+    | ./local/text2token.py -t "char" > $lang_char_dir/text
+
+  # Prepare words.txt
+  grep "\"text\":" data/manifests/aishell2_supervisions_train.jsonl.gz \
```

Comment:
Here，I think you are regarding each token as a word. Of course, it's OK. If you don't use HLG for training or testing in this recipe, the words.txt file will not influence the final results. For avoiding repeating operations, maybe you can use the following command:
`cp -r $lang_char_dir/text $lang_char_dir/text_words`

BTW, you also can refer to the following commands if you think it's necessary:
```
  # Prepare text normalize
  python ./local/text_normalize.py \
    --input $lang_char_dir/text_full \
    --output $lang_char_dir/text

  # Prepare words segments
  python ./local/text2segments.py \
    --input $lang_char_dir/text \
    --output $lang_char_dir/text_words_segmentation

  cat $lang_char_dir/text_words_segmentation | sed "s/ /\n/g" \
    | sort -u | sed "/^$/d" \
    | uniq > $lang_char_dir/words_no_ids.txt
```

Reply:
Oh, sorry, I notice that you have changed it. OK, you can ignore this comment.